testing/clair.go

```diff
@@ -0,0 +1,83 @@
+package vtesting
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	v1 "github.com/coreos/clair/api/v1"
+)
+
+const (
+	jsonContentType = "application/json;charset=utf-8"
+	basicUsername   = "shopifolk"
+	basicPassword   = "shopify"
+)
+
+type clairAPIMock struct {
+	digestLayers map[string]bool
+}
+
+func (mock *clairAPIMock) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
+	user, pass, _ := req.BasicAuth()
+	if user != basicUsername || pass != basicPassword {
+		http.Error(writer, fmt.Sprintf("Unauthorized: %s", req.URL.Path), 401)
+	}
+	switch req.URL.Path {
+	case "/v1/layers":
+		var data map[string]v1.Layer
+		_ = json.NewDecoder(req.Body).Decode(&data)
+		mock.digestLayers[data["Layer"].Name] = true
+		fmt.Print(data["Layer"].Name)
+
+		respond(writer, jsonContentType, "okay")
+		return
+	default:
+		for key := range mock.digestLayers {
+			if "/v1/layers/"+key == req.URL.Path {
+				respond(writer, jsonContentType, newClairLayerWithVulnerabilities())
+				return
+			}
+		}
+	}
+	fmt.Print(req.URL.Path)
+	http.Error(writer, fmt.Sprintf("failed to handle request: %s", req.URL.Path), 500)
+}
+
+// NewTestClairServer creates a mock of Clair
+func NewTestClairServer(t *testing.T) *httptest.Server {
+	handler := new(clairAPIMock)
+	handler.digestLayers = map[string]bool{}
+
+	server := httptest.NewServer(handler)
+	return server
+}
+
+func newClairLayerWithVulnerabilities() v1.LayerEnvelope {
+	layerEnv := v1.LayerEnvelope{
+		Layer: &v1.Layer{
+			Features: []v1.Feature{
+				{
+					Vulnerabilities: []v1.Vulnerability{
+						{
+							Name:        "bad vul 1",
+							Description: "some bad vul that i cant comprepend",
+							FixedBy:     "Hackerman",
+							Severity:    "Medium",
+						},
+						{
+							Name:        "bad vul 2",
+							Description: "Dont know what this is",
+							FixedBy:     "Hackerman",
```

Comment:
```suggestion
							FixedBy:     "Hackerwoman",
```

So that we don't have only one vulnerability fixer! 😜

Reply:
I also added Hackerdog and Hackercat haha